classes/test/engines/concurrent.php

```diff
@@ -132,9 +132,34 @@ public function run(atoum\test $test)
 				$phpCode .= '$test->getMockGenerator()->disallowUndefinedMethodUsage();';
 			}
 
-			foreach ($test->getExtensions() as $extension)
+			$extensions = $test->getExtensions();
+
+			foreach ($extensions as $extension)
 			{
-				$phpCode .= '$test->addExtension(new ' . get_class($extension) . ');';
+				$configuration = $exception = null;
+				$extensionClass = get_class($extension);
+
+				try
+				{
+					$configuration = get_class($extensions[$extension]);
```

Comment:
$extensions[$extension] is null, then get_class returns the current class, and then the test on line 148 is false.

writting  had fixed the problem :
```
$configurationObject = $extensions->getInfo();
                    if (null !== $configurationObject) {
                        $configuration = get_class($configuration);
                    }
```